### PR TITLE
Some whisper code refactor.

### PR DESCRIPTION
**Pull request by eswordthecat**
Make some object hear your lewd whispers! (in a distance)

Possible situation:
Sec officer whispers something to you (secrets, etc.. you name it!)?
Now If you have bring tape recorder (assume it's recording) at your pocket you can now black mail him/her forever!

Fix something about whisper.

### CODE REVIEW

**Comment by eswordthecat** [comment]
Can be pulled if success.

**Comment by eswordthecat** [comment]
Okay now for realsies, it's ready for pull

File: code/modules/mob/living/carbon/human/whisper.dm [hunk 1]
```diff
@@ -44,34 +43,37 @@
 		said_last_words = src.stat
 	message = treat_message(message)
 
-	var/list/listening_dead = list()
-	for(var/mob/M in player_list)
-		if(M.stat == DEAD && M.client && ((M.client.prefs.toggles & CHAT_GHOSTEARS) || (get_dist(M, src) <= 7)))
-			listening_dead |= M
+	var/listeners = get_hearers_in_view(1, src) | observers
```
Comment (ComicIronic) [inline code/modules/mob/living/carbon/human/whisper.dm]: Why did you remove ghostears functionality.

Comment (eswordthecat) [inline code/modules/mob/living/carbon/human/whisper.dm]: observers = ghost

GHOST_EARS is handle through Hear() of observers.
